plugins/modules/mysql_user.py

```diff
@@ -365,7 +365,7 @@ def main():
     ssl_key = module.params["client_key"]
     ssl_ca = module.params["ca_cert"]
     check_hostname = module.params["check_hostname"]
-    db = ''
+    db = 'mysql'
```

Comment:
@d-rupp have you checked if `db` was really used in queries ?
Have you tested the module on your side to check if it really solves your issue ?

You can place a copy of the module in a `library` dir sitting next to your playbook to try.

Reply:
Well, i did test the fix forward and backwards. 
I changed the file on my disk in the collection itself (```/home/vagrant/.local/lib/python3.8/site-packages/ansible_collections/community/mysql/plugins/modules/mysql_user.py``` in my case).

With the fix i applied a user creation, modification and deletion and replication on the filtered node kept running.

Without the fix it breaks with ```Last_SQL_Error: Error 'There is no such grant defined for user 'foo' on host '10.10.10.1'' on query. Default database: ''. Query: 'REVOKE PROCESS,REPLICATION CLIENT ON *.* FROM 'foo'@'10.10.10.1''```

Reply:
Small error: i apply a modification to an existing user, a creation of a new user, a modification to the new user and deletion of the new user in that order.
The first modification of the existing user already breaks replication as the users permissions are not present on the filtered replica.
Now if i run the test again (without the fix) without modifying the existing user, but creating a new user -> the change will get applied to the replica even though it should NOT be applied and filtered out.